solutions/platinum/poi-10-frog.mdx

```diff
@@ -2,96 +2,171 @@
 id: poi-10-frog
 source: POI 2010
 title: Frog
-author: Benjamin Qi
+author: Benjamin Qi, Justin Ji
 ---
 
+## Explanation
+
+To calculate the $k$-th closest rock from a certain location,
+we maintain a sliding window of size $k+1$. Essentially, we sweep
+left to right, and keep on moving the window right until either:
+
+1. The window cannot be moved right any further.
+2. Moving the window to the right would increase the maximum distance from this current location.
+
+Afterwards, we can proceed to binary lift to get our answer. However, directly calculating binary lifts
+the traditional way would take up too much memory. From here, there are two ways to proceed:
+
+## Implementation 1
+
+Iterate over the power of two we are lifting on in increasing order. Similar to binary exponentiation,
```

Comment:
might want to mention that a [linear memory binary lifting](https://codeforces.com/blog/entry/74847) exists so we can answer queries online